### PR TITLE
[10.0][IMP] account_asset_disposal: disposal move with expenses

### PR DESCRIPTION
**Pull request by luismontalba**
This PR adds an expense account at asset category level to post expenses when
asset is disposed.

Based on #384

cc @Tecnativa

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
I have improved the PR, but my approval for the rest.

@carlosdauden @rafaelbn please review

File: account_asset_disposal/models/account_asset_category.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+from odoo import models, fields
+
+
+class AccountAssetCategory(models.Model):
+    _inherit = "account.asset.category"
+
+    account_loss_id = fields.Many2one(
+        comodel_name="account.account", string="Loss Account",
+        domain=[('type', '=', 'other')],
```
Comment (carlosdauden) [inline account_asset_disposal/models/account_asset_category.py]: internal_type instead of type

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: account_asset_disposal/models/account_asset_category.py (hunk 1 above)
Comment (pedrobaeza) [reply account_asset_disposal/models/account_asset_category.py]: Ouch, I have forgotten to remove that one. I have removed the other domain in the wizard.

Comment (pedrobaeza) [reply account_asset_disposal/models/account_asset_category.py]: Changed

File: account_asset_disposal/models/account_asset_asset.py [hunk 2]
```diff
@@ -14,48 +15,152 @@ class AccountAssetAsset(models.Model):
...
+            'date': date,
+            'debit': depreciation_value,
+            'credit': 0.0,
+        }
 
-    def get_disposal_date(self):
-        return fields.Date.context_today(self)
+    def _disposal_line_loss_prepare(self, date, loss_account):
+        self.ensure_one()
+        loss_value = self.salvage_value + self.value_residual
```
Comment (carlosdauden) [inline account_asset_disposal/models/account_asset_asset.py]: I have some doubt with this, but they are beyond the scope of this module.

**Review by carlosdauden — APPROVED** [review]

File: account_asset_disposal/models/account_asset_asset.py (hunk 2 above)
Comment (pedrobaeza) [reply account_asset_disposal/models/account_asset_asset.py]: It was validated by the customer.

**Comment by luismontalba** [comment]
Changes done

**Pull request merged**